config/ompi_check_ucx.m4

```diff
@@ -104,44 +104,56 @@ AC_DEFUN([OMPI_CHECK_UCX],[
-                                 [AC_DEFINE([HAVE_UCP_TAG_SEND_NBR],[1],
-                                            [have ucp_tag_send_nbr()])], [],
-                                 [#include <ucp/api/ucp.h>])
-                  AC_CHECK_DECLS([ucp_ep_flush_nb, ucp_worker_flush_nb,
-                                  ucp_request_check_status, ucp_put_nb, ucp_get_nb,
-                                  ucp_put_nbx, ucp_get_nbx, ucp_atomic_op_nbx],
-                                 [], [],
-                                 [#include <ucp/api/ucp.h>])
-                  AC_CHECK_DECLS([ucm_test_events,
-                                  ucm_test_external_events],
-                                 [], [],
-                                 [#include <ucm/api/ucm.h>])
-                  AC_CHECK_DECLS([UCP_ATOMIC_POST_OP_AND,
-                                  UCP_ATOMIC_POST_OP_OR,
-                                  UCP_ATOMIC_POST_OP_XOR,
-                                  UCP_ATOMIC_FETCH_OP_FAND,
-                                  UCP_ATOMIC_FETCH_OP_FOR,
-                                  UCP_ATOMIC_FETCH_OP_FXOR,
-                                  UCP_PARAM_FIELD_ESTIMATED_NUM_PPN],
-                                 [], [],
-                                 [#include <ucp/api/ucp.h>])
-                  AC_CHECK_DECLS([UCP_WORKER_ATTR_FIELD_ADDRESS_FLAGS],
-                                 [AC_DEFINE([HAVE_UCP_WORKER_ADDRESS_FLAGS], [1],
-                                            [have worker address attribute])], [],
-                                 [#include <ucp/api/ucp.h>])
-                  AC_CHECK_DECLS([UCP_ATTR_FIELD_MEMORY_TYPES],
-                                 [AC_DEFINE([HAVE_UCP_ATTR_MEMORY_TYPES], [1],
-                                            [have memory types attribute])], [],
-                                 [#include <ucp/api/ucp.h>])
-                  AC_CHECK_DECLS([ucp_tag_send_nbx,
-                                  ucp_tag_send_sync_nbx,
-                                  ucp_tag_recv_nbx],
-                                 [], [],
-                                 [#include <ucp/api/ucp.h>])
-                  AC_CHECK_TYPES([ucp_request_param_t],
-                                 [], [],
-                                 [[#include <ucp/api/ucp.h>]])
+                  # Turn off UCX version v1.8 due to issue #8321
+                  AC_MSG_CHECKING([UCX version])
+                  AC_PREPROC_IFELSE([AC_LANG_PROGRAM([#include <ucp/api/ucp_version.h>
+                                                      #if (UCP_API_MAJOR == 1) && (UCP_API_MINOR == 8)
+                                                      #error "Invalid version"
+                                                      #endif], [])],
+                                    [AC_MSG_RESULT([ok (not 1.8.x)])],
+                                    [AC_MSG_RESULT([bad (1.8.x)])
+                                     ompi_check_ucx_happy=no])
```

Comment:
I'd also throw in this after the `AC_MSG_RESULT([bad...`:
```
AC_MSG_WARN([UCX support skipped because libucx v1.8.x was found; there is a known catastrophic bug in UCX v1.8.x])
```

Verbose is good -- it tells the user right then and there *why* we skipped UCX support, even though it was found on the system.  This, therefore, prevents the user from contacting NVIDIA support and/or the Open MPI mailing list.